django_sharding_library/router.py

```diff
@@ -31,9 +34,18 @@ def db_for_read(self, model, **hints):
             return specific_database
 
         if self.get_shard_group_if_sharded_or_none(model):
+            shard = None
```

Comment:
this code looks reasonable prime for refactoring, this look slike ti should work ,without having run it
```python

    def _get_shard(self, model, **hints):
        if self.get_shard_group_if_sharded_or_none(model):
            shard = None
            instance = hints.get('instance')
            shard_field_id = hints.get('exact_lookups', {}).get(
                getattr(model, 'django_sharding__sharded_by_field', None), None)

            if not shard_field_id and instance:
                shard_field_id = getattr(instance, getattr(model, 'django_sharding__sharded_by_field', None), None)

            if instance:
                shard = self.get_shard_for_instance(instance)
            if not shard and shard_field_id:
                shard = self.get_shard_for_id_field(model, shard_field_id)

            return shard
        return None

    def db_for_read(self, model, **hints):
        specific_database = self.get_specific_database_or_none(model)
        if specific_database:
            return specific_database

        shard = self._get_shard(model, **hints)
        if shard:
            # TODO: remove the second, should not use the shard_group attribute anywhere anymore
            shard_group = getattr(model, 'django_sharding__shard_group', getattr(model, 'shard_group', None))
            if not shard_group:
                raise Exception('Unable to identify the shard_group for the {} model'.format(model))
            routing_strategy = self.get_read_db_routing_strategy(shard_group)
            return routing_strategy.pick_read_db(shard)
        return None

    def db_for_write(self, model, **hints):
        specific_database = self.get_specific_database_or_none(model)
        if specific_database:
            return specific_database

        shard = self._get_shard(model, **hints)
        if shard:
            db_config = settings.DATABASES[shard]
            return db_config.get('PRIMARY', shard)
        return None
```